plugins/ltac/profile_ltac.ml

```diff
@@ -439,8 +439,6 @@ let print_results_filter ~cutoff ~filter =
   (* The STM doesn't provide yet a proper document query and traversal
      API, thus we need to re-check if some states are current anymore
      (due to backtracking) using the `state_of_id` API. *)
-  let valid (did,id) _ = Stm.(state_of_id ~doc:(get_doc did) id) <> `Expired in
-  data := SM.filter valid !data;
```

Comment:
What happened to this? Should the comment also be dropped?

Reply:
This is part of a commit labelled "bug".

It is the only dependency of code above toplevel on the STM. This PR exists because I want to be able to not link the STM, so I had to remove it. I can do as I did for `par:` which would be that if the STM is not linked, you get a "degraded" feature. In this case it means that if you are using ltac profiling in interactive mode and you ump back and forth, the stats printed (eg at_exit) would be cumulative of all runs, not just of the "most recent one".

I'll make a separate PR as well if you prefer.

Reply:
I don't know the ltac profiler enough to know how impactful that would be (are cumulative stats even a bad thing?)
As long as the comment is changed to avoid implying that checking is done I'm OK with it, maybe @coq/ltac-maintainers would like to comment.

Reply:
Cumulative stats are a bad thing, it means that profiling is not synchronous with the document state.  In coqtop interactive mode at least this should keep working.  I'm fine with other IDEs getting a degraded mode, but please emit a warning here when in this degraded mode.  Ideally it should be possible for other IDEs that support backtracking to have some way of doing/allowing/registering this synchronization